service/state/core_access.go

```diff
@@ -118,3 +137,22 @@ func (ca *CoreAccessor) SubmitTxWithBroadcastMode(
 	}
 	return txResp.TxResponse, nil
 }
+
+func (ca *CoreAccessor) Transfer(
+	ctx context.Context,
+	to Address,
+	amount Int,
+	gasLim uint64,
+) (*TxResponse, error) {
+	addr, err := ca.signer.GetSignerInfo().GetAddress()
+	if err != nil {
+		return nil, err
+	}
+	coins := types.NewCoins(types.NewCoin(app.BondDenom, amount))
+	msg := banktypes.NewMsgSend(addr, to.(types.AccAddress), coins)
```

Comment:
This can panic. Pls do type assertion with `ok` check as originally suggested and return an error if type is not compatible

Reply:
done